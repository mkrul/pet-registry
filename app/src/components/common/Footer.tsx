import NavLink from "../common/NavLink";

const Footer = () => {
  const currentYear = new Date().getFullYear();

  return (
    <div className="bg-base-200 py-6">
      <div className="container mx-auto px-4 text-center">
        <div className="flex justify-center space-x-6">
          <NavLink linkTo="/reports">Reports</NavLink>
          <NavLink linkTo="#">About</NavLink>
          <NavLink linkTo="#">Contact</NavLink>
          <NavLink linkTo="/reports">Search</NavLink>
        </div>
        <div className="mt-4 text-sm text-gray-500">
          <p>&copy; {currentYear} Lost Pet Registry. All rights reserved.</p>
        </div>
      </div>
    </div>
  );
};

export default Footer;
