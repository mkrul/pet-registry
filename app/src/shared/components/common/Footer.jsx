import NavLink from "./NavLink";
import { useSelector } from "react-redux";
const Footer = () => {
  const currentYear = new Date().getFullYear();
  const isAuthenticated = useSelector((state) => !!state.auth.user);

  return (
    <div className="bg-base-200 py-6" role="contentinfo" data-testid="footer">
      <div className="container mx-auto px-4 text-center">
        <div className="flex justify-center space-x-6">
          {isAuthenticated && (
            <NavLink linkTo="/dashboard/reports?action=create">New Report</NavLink>
          )}
          <NavLink linkTo="/reports">Search</NavLink>
          <NavLink linkTo="/about">About</NavLink>
          <NavLink linkTo="#">Contact</NavLink>
        </div>
        <div className="mt-4 text-sm text-gray-500">
          <p>&copy; {currentYear} Lost Pets Registry. All rights reserved.</p>
        </div>

        <div className="mt-4 text-sm text-gray-500">
          This project runs on the generosity of others. Please consider donating to help lost pets find their way home.
        </div>
      </div>
    </div>
  );
};

export default Footer;
