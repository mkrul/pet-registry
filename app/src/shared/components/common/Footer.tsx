import NavLink from "./NavLink";
import { useSelector } from "react-redux";
import { RootState } from "../../../store/store";

const Footer = () => {
  const currentYear = new Date().getFullYear();
  const isAuthenticated = useSelector((state: RootState) => !!state.auth.user);

  return (
    <div className="bg-base-200 py-6" role="contentinfo" data-testid="footer">
      <div className="container mx-auto px-4 text-center">
        <div className="flex justify-center space-x-6">
          {isAuthenticated && (
            <NavLink linkTo="/dashboard?section=reports&action=create">New Report</NavLink>
          )}
          <NavLink linkTo="/reports">Search</NavLink>
          <NavLink linkTo="#">About</NavLink>
          <NavLink linkTo="#">Contact</NavLink>
        </div>
        <div className="mt-4 text-sm text-gray-500">
          <p>&copy; {currentYear} Lost Pets Registry. All rights reserved. ðŸ‡µðŸ‡¸ Free Palestine.</p>
        </div>

        <div className="mt-4 text-sm text-gray-500">
          This project depends on the generosity of others to keep it running. Please consider donating to support its continued development.
        </div>
      </div>
    </div>
  );
};

export default Footer;
