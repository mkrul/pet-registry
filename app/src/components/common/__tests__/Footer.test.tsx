import { describe, it, expect, vi, beforeEach, beforeAll, afterAll } from "vitest";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/vitest";
import { MemoryRouter } from "react-router-dom";
import Footer from "../Footer";

vi.mock("../NavLink", () => ({
  default: ({ children, linkTo }: { children: React.ReactNode; linkTo: string }) => (
    <a href={linkTo} data-testid="nav-link">
      {children}
    </a>
  )
}));

describe("Footer", () => {
  const currentYear = new Date().getFullYear();

  beforeAll(() => {
    vi.setSystemTime(new Date(currentYear, 0, 1));
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders footer with current year", () => {
    render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    const copyrightText = screen.getByText(/Lost Pet Registry/);
    expect(copyrightText).toBeDefined();
    expect(copyrightText.textContent).toContain(currentYear.toString());
    expect(copyrightText.textContent).toContain("All rights reserved");
  });

  it("renders all navigation links", () => {
    render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    const navLinks = screen.getAllByTestId("nav-link");
    const expectedLinks = [
      { text: "Reports", path: "/reports" },
      { text: "About", path: "#" },
      { text: "Contact", path: "#" },
      { text: "Search", path: "/reports" }
    ];

    expect(navLinks).toHaveLength(expectedLinks.length);

    navLinks.forEach((link, index) => {
      expect(link).toHaveAttribute("href", expectedLinks[index].path);
      expect(link).toHaveTextContent(expectedLinks[index].text);
    });
  });

  it("applies correct styling classes", () => {
    render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    const footer = screen.getByRole("contentinfo");
    expect(footer).toHaveClass("bg-base-200", "py-6");

    const container = footer.firstChild;
    expect(container).toHaveClass("container", "mx-auto", "px-4", "text-center");

    const copyrightDiv = screen.getByText(/Lost Pet Registry/).parentElement;
    expect(copyrightDiv).toHaveClass("mt-4", "text-sm", "text-gray-500");
  });

  it("matches snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    const yearElement = container.querySelector("p");
    if (yearElement) {
      yearElement.textContent =
        yearElement.textContent?.replace(currentYear.toString(), "YEAR") ?? null;
    }

    expect(container).toMatchSnapshot();
  });
});
