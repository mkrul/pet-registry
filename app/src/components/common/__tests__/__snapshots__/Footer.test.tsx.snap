// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > matches snapshot 1`] = `
<div>
  <div
    class="bg-base-200 py-6"
    role="contentinfo"
  >
    <div
      class="container mx-auto px-4 text-center"
    >
      <div
        class="flex justify-center space-x-6"
      >
        <a
          data-testid="nav-link"
          href="/reports"
        >
          Reports
        </a>
        <a
          data-testid="nav-link"
          href="#"
        >
          About
        </a>
        <a
          data-testid="nav-link"
          href="#"
        >
          Contact
        </a>
        <a
          data-testid="nav-link"
          href="/reports"
        >
          Search
        </a>
      </div>
      <div
        class="mt-4 text-sm text-gray-500"
      >
        <p>
          Â© YEAR Lost Pet Registry. All rights reserved.
        </p>
      </div>
    </div>
  </div>
</div>
`;
